lib/instrumentation/modules/fastify.js

```diff
@@ -0,0 +1,31 @@
+'use strict'
+
+const semver = require('semver')
+
+module.exports = function (fastify, agent, version, enabled) {
+  agent.logger.debug('wrapping fastify build function')
+
+  return wrappedBuild
+
+  function wrappedBuild () {
+    const _fastify = fastify.apply(null, arguments)
+
+    agent.logger.debug('adding onRequest hook fo fastify')
+
+    if (semver.gte(version, '2.0.0')) {
+      _fastify.addHook('onRequest', (req, reply, next) => {
+        const name = req.method + ' ' + reply.context.config.url
```

Comment:
Could simplify by just doing `(reply.context || reply._context)` to support all versions with one handler.

Reply:
Doh, great idea 😄